cloudsync/tests/test_provider.py

```diff
@@ -266,20 +297,17 @@ def set_root(self, root_path: str = None, root_oid: str = None):
     def info_oid(self, oid: str, use_cache=True) -> Optional[OInfo]:
         return self.__strip_root(self.prov.info_oid(oid))
 
-    @wrap_retry
```

Comment:
removed all retries for listdir.... why?

Reply:
I did put a comment, as well new code, into wrap_retry. wrap_retry does nothing for methods that are generators. The retry code happens on the call to the function which returns a generator, and doesn't actually run any code in listdir at that time. Then you iterate over the generator, outside of `wrap_retry` and that's when listdir actually starts running, so no retrying will actually happen. I put an assertion into wrap_retry to guarantee you don't get a false sense of security, thinking that the wrapper was going to help you when it won't

Reply:
I will play around with fixing `wrap_retry` so that it will work on generators as well as regular methods, and if so I'll put the `wrap_retry` back on `listdir`